pkg/server/commands/connectedobjects/connected_objects.go

```diff
@@ -225,9 +243,12 @@ func (c *ConnectedObjectsQuery) execute(
 	subg, subgctx := errgroup.WithContext(ctx)
 	subg.SetLimit(int(c.resolveNodeBreadthLimit))
 
-	for i, ingress := range ingresses {
-		span.SetAttributes(attribute.String(fmt.Sprintf("_ingress %d", i), ingress.String()))
+	for _, ingress := range ingresses {
 		innerLoopIngress := ingress
+		if req.Ingress != nil && req.Ingress.Condition == graph.RequiresFurtherEvalCondition {
+			// propagate the condition to upcoming reverse expansions
+			innerLoopIngress.Condition = graph.RequiresFurtherEvalCondition
```

Comment:
The ingresses status should come directly from the ingress itself. I'm confused why we are mutating ingress conditions in the expansion paths?

Could you elaborate?

Reply:
Talked about this offline. We discussed moving this to propagating a boolean or something indicating a conditional ingress was evaluated along the evaluation path.

Reply:
added a TODO

Reply:
Please create the issue and cross referencing with this PR.

Reply:
https://github.com/openfga/openfga/issues/957